os/os.proto

```diff
@@ -0,0 +1,321 @@
+// This file defines a gNOI API used for OS installation.
+syntax = "proto3";
+
+package gnoi.os;
+
+import "github.com/openconfig/gnoi/types/types.proto";
+
+option (types.gnoi_version) = "0.1.0";
+
+// The OS service provides an interface for OS installation on a Target. The
+// Client progresses through 3 RPCs:
+//   1) Installation - provide the Target with the OS package.
+//   2) Activation - activate an installed OS package.
+//   3) Verification - verify that the Activation was successful.
+//
+// Dual Supervisor Target is supported, where the above process is executed once
+// for each Supervisor.
+//
+// Note that certain platforms may have particular approaches to upgrade the
+// firmware of specific components, eg., power supply units, etc.. In addition,
+// platforms may have processes to apply patches to the running OS. Handling
+// these exceptions introduces extra complexities. For Targets that implement
+// this service, component firmware upgrade or OS patching must be embedded
+// within an OS upgrade.
+service OS {
+  // Install transfers an OS package into the Target. No concurrent Install RPCs
```

Comment:
Is there a common way that a failure due to not allowing concurrent requests should be signaled?

Reply:
Yes, line 232: InstallResponse->InstallError->INSTALL_IN_PROGRESS